### PR TITLE
[TS] LPS-173068

### PR DESCRIPTION
**Pull request by ericyanLr**
Hi Team,

Can you help review this PR? If this is not the right team for this review, please help advise which team to send to.

Notes from @noornajj: 

> https://issues.liferay.com/browse/LPS-173068
> 
> The issue here is that a null literal is added to a search summary when a repeatable field is empty and there is at least one other field that is not empty. This occurs because repeatable fields are returned as arrays and when the field is empty, an empty string is added to the array --> [""] 
> _getSortableValue will return the empty string as null and then null is added to the string bundler as a literal. 
> 
> To fix the issue, I had _getSortableValue return an empty string instead of null.

Please let us know if you have any questions.
Thanks!

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java [hunk 1]
```diff
@@ -372,7 +372,9 @@ public String extractIndexableAttributes(
 
 				Serializable value = field.getValue(locale);
 
-				if (value == null) {
+				if ((value == null) ||
+					Objects.equals(value, StringPool.BLANK)) {
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:376]: @ericyanLr @noornajj why don't you use `Validator.isBlank(value)` instead of `Objects.equals(value, StringPool.BLANK)`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java [hunk 2]
```diff
@@ -372,7 +372,9 @@ public String extractIndexableAttributes(
 
 				Serializable value = field.getValue(locale);
 
-				if (Validator.isNull(value)) {
+				if ((value == null) ||
+					Objects.equals(value, StringPool.BLANK)) {
+
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:377]: @ericyanLr @noornajj why did you have to change from 
```
Validator.isNull(value)
``` 
to 
```
(value == null) || Objects.equals(value, StringPool.BLANK)
``` 

if `Validator.isNull(value)` returns `true` if value is `null` or `blank`? see [here](https://github.com/liferay/liferay-portal/blob/faf682e989d8872bdea67d3b17c2d4318c888502/portal-kernel/src/com/liferay/portal/kernel/util/Validator.java#L915-L919)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java [hunk 3]
```diff
@@ -810,7 +816,7 @@ private String _getSortableFieldName(String name) {
 	private String _getSortableValue(
 		DDMFormField ddmFormField, Locale locale, Serializable value) {
 
-		if (Validator.isNull(value)) {
+		if (value == null) {
 			return null;
 		}
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:821]: @noornajj @ericyanLr based on your comment:

>This occurs because repeatable fields are returned as arrays and when the field is empty, an empty string is added to the array --> [""]
> _getSortableValue will return the empty string as null and then null is added to the string bundler as a literal.
> 
> To fix the issue, I had _getSortableValue return an empty string instead of null.

this change would be enough to fix the bug. Why did you need the other changes?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java (hunk 1 above)
Comment (noornajj) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:376]: Hi @carolmariaabb 
Because value is serializable and isBlank() takes in a string value

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java (hunk 2 above)
Comment (noornajj) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:377]: This is to account for "null" literals. If `Validator.isNull(value)` is used, then if the field's value is "null" it'll be skipped

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java (hunk 3 above)
Comment (noornajj) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:821]: The other changes were to account for extra spaces that were added to the string bundler when the value was an empty string

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java (hunk 2 above)
Comment (ericyanLr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:377]: @carolmariaabb, adding on to Noor's comment, do you know if "null" literals should be skipped or retained? This was a topic we weren't exactly sure about.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java (hunk 1 above)
Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:376]: @noornajj can we change to `Validator.isBlank(String.valueOf(value))`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java (hunk 2 above)
Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:377]: @ericyanLr @noornajj In my opinion, we can skip "null" literals, because it seems like an edge case to me.. But you can ask in the #help-forms-data-engine slack channel to confirm.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java (hunk 1 above)
Comment (noornajj) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:376]: Sure @carolmariaabb I can change it to that

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java (hunk 2 above)
Comment (noornajj) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:377]: @carolmariaabb @ericyanLr I have posted the question in the channel and waiting for a reply

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java (hunk 1 above)
Comment (noornajj) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:376]: Hello @carolmariaabb 
Should I just make this change and resend the PR?

Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:376]: Yes @noornajj

Comment (noornajj) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java:376]: @carolmariaabb new PR sent https://github.com/liferay-forms/liferay-portal/pull/2127
